build.sbt

```diff
@@ -211,6 +212,43 @@ lazy val testStdLayout = Project("daffodil-test-stdLayout", file("test-stdLayout
   .dependsOn(tdmlProc % "test")
   .settings(commonSettings, nopublish)
 
+/* Workaround: certain reflection (used by JAXB) isn't allowed by default in JDK 17:
+ * https://docs.oracle.com/en/java/javase/17/migrate/migrating-jdk-8-later-jdk-releases.html#GUID-7BB28E4D-99B3-4078-BDC4-FC24180CE82B
+ *
+ * While we can handle this JVM quirk at build time, at runtime we won't know
+ * a user's JVM version. We'll provide documentation and an extension setting
+ * to add these flags to the extension-launched debugger backend.
+ */
+lazy val extraJvmOptions: Seq[String] =
+  if (scala.util.Properties.isJavaAtLeast("17"))
+    Seq(
+      "--add-opens",
+      "java.base/java.lang=ALL-UNNAMED",
+    )
+  else Seq()
+
+lazy val xjcSettings =
+  Seq(
+    libraryDependencies ++= Seq(
+      "com.sun.xml.bind" % "jaxb-impl" % "2.2.11",
+      "javax.activation" % "activation" % "1.1.1",
+      "org.glassfish.jaxb" % "jaxb-xjc" % "2.2.11",
+    ),
+    xjcCommandLine += "-nv",
+    xjcCommandLine += "-p",
+    xjcCommandLine += "org.apache.daffodil.tdml",
+    xjcLibs := Seq(
+      "org.glassfish.jaxb" % "jaxb-xjc" % "2.2.11",
```

Comment:
Please label which of these are the runtime dependencies, and which are the compile-time dependencies. (I'm assuming there is that separation).

Reply:
I thought the plugin defined the same set of libs, but the activation jar was [only added recently](https://github.com/sbt/sbt-xjc/commit/2178c47e50020dd09ad55781c932282250b8d752) (2021!) but hasn't been released yet. (Latest release is 0.10, which doesn't even have a tag.)

I'm testing reducing the additional library mentions in this settings block.

Reply:
Oof, I didn't notice it that change is two years old without a release. I guess we could just do this in that case to just fix the issue:
```scala
xjcLibs += "javax.activation"   % "activation" % "1.1.1",
```
Keeps things a little less verbose. And I doubt we have to worry about he plugin being updated and breaking things. Doubt there will be another release anytime soon.

Alternatively, has anyone looked at [scalaxb](https://github.com/eed3si9n/scalaxb)? It looks like it's created by the main SBT maintainer, so imagine it will continue to be maintained for a while and with SBT best practices.

I made changes this PR to use scalaxb instead of xjc and it was pretty straightforward. This was all I had to change to this PR:

```scala
lazy val tdmlLib = Project("daffodil-tdml-lib", file("daffodil-tdml-lib"))
  .enablePlugins(ScalaxbPlugin)
  .dependsOn(macroLib % "compile-internal", lib, io, io % "test->test", slf4jLogger % "test")
  .settings(commonSettings)
  .settings(scalaxbSettings)

lazy val scalaxbSettings = Seq(
  Compile / scalaxb / scalaxbPackageName := "org.apache.daffodil.tdml.scalaxb",
  Compile / scalaxb / sources := Seq(
    (lib / Compile / resourceDirectory).value / "org/apache/daffodil/xsd/tdml-core.xsd",
   )
)
```

Nicely, it doesn't auto enable so it can be enabled for only the one project we need without the `sources := Seq()` hack.

The only issue I ran into is the generator adds imports that aren't used, which Daffodil treats as an error. But disabling fatal warnings, the generated code compiles fine. There's probably a way to fix this without disabling warnings. And if not, I imagine scalaxb would accept a patch and get a release out quickly.

I also noticed it generates immuatable case classes. But it does have a `scalaxbGenerateMutable` setting to make the case members `var`s if that's preferred

Plus, I imagine it is probably more straightforward to move our existing TDML runner functionality into those generated classes since it's already scala code, which would be nice. It would probably reduce boilerplate in our existing TDML runner lib, and make it so you could run tests without having to serialize wih saxb and then reload with the TDML runner. That's a far off goal, but scalaxb makes that more possible I think.

Reply:
Re: scalaxb, I had initially attempted to implement the JAXB bindings with that plugin but was unsuccessful. I could not find a way to handle duplicate names for items in the schemas. In the SbtXjc plugin, we handled that with the bindings.xjb file, but I could not find a way to provide anything like that in scalaxb. We won't run into that issue with tdml-core.xsd, but if you want to eventually go back to importing all of the schemas in JAXB, I don't believe scalaxb would currently be able to handle that.

Reply:
> duplicate names for items in the schemas.

Can you clarify what this means? I'm not sure I understand.

I'm also not against the tdml-core split. It would just be a plus if we didn't need it with scalaxb, but I guess we do.

Regardless of xjc vs scalaxb, I prefer tdml-core over bindings.xjb.

Reply:
If two items in all imported schemas (elements, attributes, etc.) share a name, JAXB will attempt to create a binding for them at the same Java file location, which, obviously, won't work, so there needs to be a way to tell it how to rename one of them. With the SbtXjcPlugin, that is the bindings.xjb file. With the scalaxb plugin, I could not find a way to do this outside of refactoring the schemas so that everything has a unique name or by submitting a patch to the plugin to support a bindings file... Although, the tdml-core refactor also resolves this as long as we only want JAXB bindings for that subset of the TDML schema.

If you are going to eventually integrate the JAXB bindings into the TDMLRunner, I imagine that would require generating bindings for the base DFDL schemas, which do have clashing names in a few places. I'm not suggesting that we make any changes at this point - just want to point it out so that this isn't a surprise if/when you get to this point.

Reply:
I see, thanks for the explanation. Sounds like the tdml-core split is a better solution.

If we did integrate the TDMLRunner into the JAXB bindings, I'm not sure we would even want bindings for the stuff we ignore now. For example, embedded DFDL schemas just want to be treated as raw strings or XML nodes or something. Those schemas are just written to a file and compiled with Daffodil, so converting them to JAXB bindings just to serialize back to XML is probably unnecessary.

But I don't think there's a plan to integrate  the TDML runner into these bindings, at least in the near term, so whatever the solution is for duplicate identifiers, it can be dealt with then.